### DISCUSSION TITLE
Support the use of `$slot` in Livewire components⚡️

### DISCUSSION THREAD

**Discussion by ralphjsmit** [livewire/livewire#4199] [3. Feature ideas]
Hey Livewire team,

It would be great if you could do something like this:

```
<x-livewire:foo-bar>
    @livewire('other-component')
</x-livewire:foo-bar />
```

Is this supported? If not, would it take much to support it?

**Comment by joshhanley** [comment]
@ralphjsmit currently it's not supported and I don't think it would be simple to implement but it would be interesting if we could!

So my question for you would be, how would you expect it to work?

Imagine a scenario where you have a component that can take slot content, on the initial render, it would happily display the slot (who's contents would be passed down from a parent component or from a standard blade view served from a controller).

But what happens on the next render?

So our component makes a request to the server, and tries to render itself (unlike before where it was rendered by the parent with slot passed in), but where does it get the slot content from this time?

I don't have a good answer to that last question, but would love to know if you have any ideas?

Hope this helps!

**Reply by joshhanley** [reply]
@walirazzaq I thought I saw a tweet about you experimenting with this 👀

Any chance you can share how you are implementing it?

**Reply by ralphjsmit** [reply]
Yeah, okay, I understand the problem. A solution would be to only use other templates (like `@include()`), but if that would be the only use case, it's still not very useful.

What about temporary caching the unrendered slot content and re-rendering the slot from the cache?

**Reply by walirazzaq** [reply]
@joshhanley I'll be more than happy to share 👍🏼. It's wasn't that simple as I thought before starting but I have covered plenty of stuff so far.
I have been on it as "Make it work" strategy so code itself is a bit dirty as of now. 
I can give you access to repo and a walk through of source code if you are available.

**Reply by joshhanley** [reply]
@walirazzaq no rush! Share when you are ready to share 🙂

But I'd be curious as to what approach you have taken if you can give a TL;DR to get us all on the same page 😁

**Reply by walirazzaq** [reply]
I am using same approach as used in actual laravel blade component and slots and caching for subsequent request (subsequent requests are giving tough times) 😀
When try to cover on use case, other breaks. Latest commit message tells the story 😆 
![image](https://user-images.githubusercontent.com/90649496/142784075-7b06ec6d-85a7-4bc0-ad48-804b6cceea0c.png)

**Reply by joshhanley** [reply]
Ah yep makes sense. Yeah it's a tough problem to try and solve!

**Comment by rainsens** [comment]
\<livewire:table.table>
    \<livewire:table.title>Title</livewire:table.title>
    \<livewire:table.body>Body</livewire:table.body>
</livewire:table.table>

A really useful feature, when we abstract frontend into function.

**Comment by AurelienMendes** [comment]
Are the slots components still in the roadmap ? 
I'm afraid not and I still don't understand such a choice not to implement this feature in Livewire 3.

**Reply by lrljoe** [reply]
Not sure I really understand the use case, what're you looking to do and for what purpose.

**Reply by AurelienMendes** [reply]
Hi @lrljoe,

Let me show you one of my use case.
Suppose you have an UI framwork, with a Table component and you want to write your own component "**model-list**", including this Table one, to be able to list a given Model.

You have a **Route** which return an **index view** and in this view you have something like that:

`<livewire:app.model-list :model="\App\Models\Country::class" :striped="false" />`

In your model-list component, you have, 

- the Table component from your UI framework, 
- but you also add a Header (also from your UI) which include a title, and a search input, filters and a Add button

All of this work perfectly, many of properties are defined in your Country Model (like a searchable field list, a list of actions, the list of columns to display in the Table, ...), and you are very happy since you can plug it to every Model easily.

But, one day, someone tells you that it would be great to add some statistics in your list pages, between the Header, and the Table itself.

There, since it's very specific to each Model to list, it would be great to be able to do:

`<livewire:app.model-list :model="\App\Models\Country::class" :striped="false">`
`    <template slot="statistics">`
`        <div class="flex">`
`            <div id="stat1">...</div>`
`            <div id="stat2">...</div>`
`            ...`
`        </div>`
`    </template>`
`</livewire>`


So If you have an another (and why not better) solution for this use case I'm all ears

Thanks !

**Reply by lrljoe** [reply]
Personally, I have

A landing view/component

The blade for that extends a template with slots

One slot is for the header

Then I add the content...

e.g. a Livewire table component which is a rappasoft table
If I want to add content before/after the table, then I just pass the blade path into it and use the rappasoft configurable areas.

Or I add it to the parent view/component blade

What you're suggesting doesn't quite make sense.
